### PR TITLE
[INF47]  Initialization of Sphinx based documentation

### PR DESCRIPTION
**Pull request by KedoKudo**
- Original Gitlab task: [INF102](https://code.ornl.gov/sns-hfir-scse/infrastructure/web-monitor/-/issues/102)

This PR introduces the following changes:
- add the overall skeleton structure for a Sphinx-based documentation
- archive some legacy documents
- add auto generated module index
  - auto generated page for `settings.py` for each app is removed

After building the documentation in the `docs` folder with `make html`, the index page should looks like the following

image

Known issues:
- ~~Missing django settings during build stage leads to various warnings.~~
- The user cases in markdown cannot be included in the index page.
- Various warnings caused by incorrect index in docstring.

### CODE REVIEW

**Comment by peterfpeterson** [comment]
Comments on the remaining items
* The django settings can be fixed using the same "trick" that is [used for unittests](https://github.com/neutrons/data_workflow/blob/next/.github/workflows/ci.yml#L28)
* There is a [sphinx plugin for markdown](https://www.sphinx-doc.org/en/master/usage/markdown.html), or we could decide to (not in this PR) convert them to rst. Converting to rst would probably be better since it'll put all of the documentation in one markup/markdown language
* bad docstrings should be a follow-on

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+
+# -- Project information -----------------------------------------------------
+
+project = "Web Monitor"
+copyright = "2022, SCSE@ORNL"
+author = "SCSE@ORNL"
+
+# The short X.Y version
+version = ""
```
Comment (peterfpeterson) [inline docs/conf.py:31]: `version` and `release` can be added to the [bumversion configuration in `setup.cfg`](https://github.com/neutrons/data_workflow/blob/next/setup.cfg#L1-L4)

File: docs/make.bat [hunk 2]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (peterfpeterson) [inline docs/make.bat:1]: I'm tempted to remove this file and not support windows

File: docs/conf.py (hunk 1 above)
Comment (KedoKudo) [reply docs/conf.py:31]: do I manually add them in `conf.py` or is there a better way to do it?

File: docs/make.bat (hunk 2 above)
Comment (KedoKudo) [reply docs/make.bat:1]: got it, I will remove the .bat file now

**Review by peterfpeterson — APPROVED** [review]

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**